recipes/mysql/build.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# this script is based off the homebrew package:
+# https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb
+
+# Seems to be needed to build in some spots.
+for pth in /lib/cpp /lib64/cpp /usr/bin/cpp
+do
+    if [ ! -e ${pth} ]
+    then
+        # move along if these commands fail
+        mkdir -p `dirname ${pth}` || :
+        ln -s `which cpp` ${pth} || :
+    fi
+done
+
+mkdir -p build
+cd build
+
+# -DINSTALL_* are relatiove to -DCMAKE_INSTALL_PREFIX
+mkdir -p ${PREFIX}/mysql
+cmake \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DMYSQL_DATADIR=${PREFIX}/mysql/datadir \
+    -DINSTALL_INCLUDEDIR=include/mysql \
+    -DINSTALL_MANDIR=share/man \
+    -DINSTALL_DOCDIR=share/doc/mysql \
+    -DINSTALL_DOCREADMEDIR=mysql \
+    -DINSTALL_INFODIR=share/info \
+    -DINSTALL_MYSQLSHAREDIR=share/mysql \
+    -DINSTALL_SUPPORTFILESDIR=mysql/support-files \
+    -DINSTALL_SCRIPTDIR=mysql/scripts \
+    -DSYSCONFDIR=${PREFIX}/etc \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_FIND_FRAMEWORK=LAST \
+    -DCMAKE_VERBOSE_MAKEFILE=OFF \
+    -Wno-dev \
+    -DWITH_UNIT_TESTS=OFF \
+    -DDEFAULT_CHARSET=utf8 \
+    -DDEFAULT_COLLATION=utf8_general_ci \
+    -DCOMPILATION_COMMENT=conda-forge \
+    -DWITH_SSL=bundled \
+    -DWITH_EDITLINE=bundled \
+    -DWITH_READLINE=bundled \
+    -DWITH_BOOST=boost \
+    -DDOWNLOAD_BOOST=1 \
+    .. &> cmake.log
+
+make
+make install &> install.log
+
+# we will run this test now and then delete the directory
+# there is no reason to ship the test dir and it is big
+# we need perl to run the test
+# if we do not have perl, then do not run the test
+if [ -x "$(command -v perl)" ]
+then
+    cd ${PREFIX}/mysql-test
+    mysql_temp_dir=`mktemp -d ${TMPDIR}/tmp/XXXXXXXXXXXX`
+    {
+        set -e
+        # the || here is a rough try...except
+        ./mysql-test-run.pl status --vardir=${mysql_temp_dir} || rm -rf ${mysql_temp_dir}
+    }
+    # always delete anything left
+    rm -rf ${mysql_temp_dir}
+    cd -
+fi
+rm -rf ${PREFIX}/mysql-test
```

Comment:
Would it make more sense to include this chunk in the `test` phase of the build instead? Could add to `run_test.sh` then `conda-build` will run this for us.

Reply:
Also should we be requiring `perl` if it is needed for testing?

Reply:
I think the idea was to not include `mysql-test` in the shipped package, since it's big. Can probably get around that with [`test/source_files`](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#source-files), though. (In that case, would also be able to ensure `perl` is present with `test/requires`.)

Reply:
Right, that's what I'm thinking as well.

Reply:
These tests do not appear to be made to run outside the build directory. We can require perl during the build and run them right away.

Reply:
So we can copy whole directories with `test/source_files` if that helps. If it doesn't, we should start requiring `perl` in `build`. Either way please pin `perl` like [so]( https://github.com/conda-forge/openblas-feedstock/blob/4a65f26c18cfa4dea9b367016855d1d1258d4384/recipe/meta.yaml#L26 ).

Reply:
I tried that. Adding perl to the build.